File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/FactoryMethod/Buttons/HtmlButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.FactoryMethod
{
    public class HtmlButton : IButton
    {
        public void OnClick()
        {
            Console.WriteLine("Click! Button says - 'Hello World!'");
        }

        public void render()
        {
            Console.WriteLine("<button>Test Button</button>");
            OnClick();
        }
    }
}

File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/factorymethod/buttons/HtmlButton.java
package com.ute.refactoring.factorymethod.buttons;

public class HtmlButton implements Button{
    @Override
    public void render() {
        System.out.println("<button>Test Button</Button>");
        onClick();
    }

    @Override
    public void onClick() {
        System.out.println("Click! Button says - 'Hello World!'");
    }
}
